storm-server/src/main/java/org/apache/storm/scheduler/Cluster.java

```diff
@@ -889,6 +889,20 @@ public double getClusterTotalMemoryResource() {
         return sum;
     }
 
+    @Override
+    public Map<String, Double> getClusterTotalGenericResource() {
+        Map<String, Double> ret = new HashMap<>();
+        for (SupervisorDetails sup : supervisors.values()) {
+            Map<String, Double> supGenericResources = sup.getTotalGenericResources();
+            for (Map.Entry<String, Double> entry : supGenericResources.entrySet()) {
+                String resourceName = entry.getKey();
+                Double amount = entry.getValue();
+                ret.put(resourceName, ret.getOrDefault(resourceName, 0.0) + amount);
```

Comment:
A stream API could make it simpler to represent aggregation of maps:.
`return supervisors.values().stream().flatMap(m -> m.getTotalGenericResources().entrySet().stream()).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, Double::sum));`

Reply:
Very good suggestions. Thank you and addressed.